### PR TITLE
gei.exe with commands to migrate GH->GH

### PR DESCRIPTION
**Pull request by dylan-smith**
#54 
#55 

Added a 2nd CLI (gei.exe) that is intended to handle GH->GH scenarios, and also some basic ADO->GH scenarios. Basically anything that can by done by only calling GH API's (including the GEI API's) and not talking directly to ADO.

Eventually this will be exposed as an extension to the official GH CLI and that will be the recommended way to consume it for customers.

The ado2gh CLI will continue to provide the more full-featured migration experience for ADO customers.

With this PR the only commands that gei.exe supports are:
- generate-script
- migrate-repo

They have GH->GH specific args and behavior that are slightly different than the ado2gh commands of the same names.

NOTE: This PR was getting big so I'm going to do separate PR's for supporting different PAT's for source/target. And a separate PR to clean up the namespace names.

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked

### CODE REVIEW

File: src/Octoshift/GithubApi.cs [hunk 1]
```diff
@@ -132,6 +142,34 @@ public virtual async Task<string> CreateMigrationSource(string orgId, string ado
             return (string)data["data"]["createMigrationSource"]["migrationSource"]["id"];
         }
 
+        public virtual async Task<string> CreateGHECMigrationSource(string orgId, string githubPat)
```
Comment (ArinGhazarian) [inline src/Octoshift/GithubApi.cs:145]: nit: `Ghec`

File: src/OctoshiftCLI.Tests/ado2gh.Commands/AddTeamToRepoCommandTests.cs [hunk 2]
```diff
@@ -2,10 +2,10 @@
 using System.CommandLine;
 using System.Threading.Tasks;
 using Moq;
-using OctoshiftCLI.Commands;
+using OctoshiftCLI.ado2gh.Commands;
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/ado2gh.Commands/AddTeamToRepoCommandTests.cs:5]: we need to address the namespaces in a separate PR I guess to be compliant with C#'s naming guidelines.

File: src/gei/Commands/GenerateScriptCommand.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.gei.Commands
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateScriptCommand.cs:9]: nit: `Gei` instead of `gei`

**Review by ArinGhazarian — COMMENTED** [review]
Overall looks good, except some nitpicks about naming methods to be compliant with C#'s [naming guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines). Same is true for namespaces and project names that should follow the naming guidelines.

File: src/OctoshiftCLI.Tests/ado2gh.Commands/AddTeamToRepoCommandTests.cs (hunk 2 above)
Comment (dylan-smith) [reply src/OctoshiftCLI.Tests/ado2gh.Commands/AddTeamToRepoCommandTests.cs:5]: when you say C# naming guidelines, is there a way to automatically enforce them in static analysis?

File: src/gei/Commands/GenerateScriptCommand.cs (hunk 3 above)
Comment (dylan-smith) [reply src/gei/Commands/GenerateScriptCommand.cs:9]: I'm going to punt this to a follow-up PR where I overhaul the namespaces

File: src/Octoshift/GithubApi.cs (hunk 1 above)
Comment (dylan-smith) [reply src/Octoshift/GithubApi.cs]: done

File: src/Octoshift/GithubApi.cs [hunk 4]
```diff
@@ -104,7 +114,7 @@ public virtual async Task<string> GetOrganizationId(string org)
             return (string)data["data"]["organization"]["id"];
         }
 
-        public virtual async Task<string> CreateMigrationSource(string orgId, string adoToken, string githubPat)
+        public virtual async Task<string> CreateADOMigrationSource(string orgId, string adoToken, string githubPat)
```
[the remark this answers is not in the document]
Comment (dylan-smith) [reply src/Octoshift/GithubApi.cs]: done

File: src/OctoshiftCLI.Tests/ado2gh.Commands/AddTeamToRepoCommandTests.cs (hunk 2 above)
Comment (ArinGhazarian) [reply src/OctoshiftCLI.Tests/ado2gh.Commands/AddTeamToRepoCommandTests.cs:5]: There could be, I need to look it up, but `Rider` warns you about those 😃

**Pull request merged**